theano/sandbox/cuda/dnn.py

```diff
@@ -1358,40 +1358,66 @@ class GpuDnnPool(DnnBase):
     ----------
     img
         The image 4d or 5d tensor.
-    desc
-        The pooling descriptor.
-
+    ws
+        Windows size.
+    stride
+        (dx, dy).
+    mode : {'max', 'average_inc_pad', 'average_exc_pad'}
+        The old deprecated name 'average' correspond to 'average_inc_pad'.
+    pad
+        (padX, padY) padding information.
+        padX is the size of the left and right borders,
+        padY is the size of the top and bottom borders.
     """
 
-    __props__ = ()
+    __props__ = ("mode",)
 
-    def make_node(self, img, desc):
-        img = as_cuda_ndarray_variable(img)
-        if not isinstance(desc.type, CDataType) \
-                or desc.type.ctype != 'cudnnPoolingDescriptor_t':
-            raise TypeError('desc must be cudnnPoolingDescriptor_t')
+    def __init__(self, mode='max'):
+        super(GpuDnnPool, self).__init__()
+        if mode == 'average':
+            mode = 'average_inc_pad'
+        assert mode in ('max', 'average_inc_pad', 'average_exc_pad')
+        self.mode = mode
 
-        if desc.owner is not None:
-            dop = desc.owner.op
-            e_ndim = dop.get_ndim() + 2  # 4 or 5
+    def prepare_node(self, node, storage_map, compute_map):
+        if len(node.inputs) == 2:
+            warnings.warn("Theano GPUDnnPoolGrad internal changed.", stacklevel=3)
+            # Old interface
+            self.mode = node.inputs[1].owner.op.mode
+            ws = theano.tensor.constant(node.inputs[1].owner.op.ws)
+            st = theano.tensor.constant(node.inputs[1].owner.op.stride)
+            pad = theano.tensor.constant(node.inputs[1].owner.op.pad)
+            node.inputs[1] = ws
+            node.inputs.append(st)
+            node.inputs.append(pad)
+            storage_map[ws] = [None]
+            storage_map[st] = [None]
+            storage_map[pad] = [None]
+            compute_map[ws] = [False]
+            compute_map[st] = [False]
+            compute_map[pad] = [False]
```

Comment:
Yes. Check if they are instance of theano.Constant and if so, set True
instead of False.

On Wed, Feb 17, 2016 at 2:05 PM, harmdevries89 <notifications@github.com>
wrote:

> In theano/sandbox/cuda/dnn.py
> <https://github.com/Theano/Theano/pull/3965#discussion_r53213581>:
>
> > +        if len(node.inputs) == 2:
> > +            warnings.warn("Theano GPUDnnPoolGrad internal changed.", stacklevel=3)
> > +            # Old interface
> > +            self.mode = node.inputs[1].owner.op.mode
> > +            ws = theano.tensor.constant(node.inputs[1].owner.op.ws)
> > +            st = theano.tensor.constant(node.inputs[1].owner.op.stride)
> > +            pad = theano.tensor.constant(node.inputs[1].owner.op.pad)
> > +            node.inputs[1] = ws
> > +            node.inputs.append(st)
> > +            node.inputs.append(pad)
> > +            storage_map[ws] = [None]
> > +            storage_map[st] = [None]
> > +            storage_map[pad] = [None]
> > +            compute_map[ws] = [False]
> > +            compute_map[st] = [False]
> > +            compute_map[pad] = [False]
>
> @nouiz <https://github.com/nouiz> Do you agree?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3965/files#r53213581>.
>